libs/media-player/src/lib/audio-annotator.tsx

```diff
@@ -0,0 +1,281 @@
+import { isNull, isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    ArrowLeft as ArrowLeftIcon,
+    ArrowRight as ArrowRightIcon,
+    Clear as ClearIcon,
+} from '@mui/icons-material/';
+import { Box, IconButton, Stack, Typography, styled } from '@mui/material';
+import { useEffect, useRef, useState } from 'react';
+import { asFormattedMediaTimecodeString } from './shared/as-formatted-media-timecode-string';
+import { AudioMIMEType } from './shared/audio-mime-type.enum';
+import { isAudioMIMEType } from './shared/is-audio-mime-type';
+import { KeyboardKey, useKeyDown } from './shared/use-key-down';
+import {
+    TimeRangeSelectionStatus,
+    TimeRangeSelectionStatusIndicator,
+} from './time-range-selection-visual';
+
+type Nullable<T> = T | null;
+
+export type MedidaPlayDirection = 'forward' | 'reverse';
+
+export enum KeyboardShortcuts {
+    play = KeyboardKey.spacebar,
+    clear = KeyboardKey.c,
+    markInPoint = KeyboardKey.i,
+    markOutPoint = KeyboardKey.o,
+    scrubForward = KeyboardKey.k,
+    scrubBackward = KeyboardKey.j,
+}
+
+const StyledAudioPlayer = styled('audio')`
+    border-radius: 20px;
+`;
+
+const calculateTimeRangeSelectionStatus = (
```

Comment:
I tried using this technique with the `errorMessage` basing its state off `inPointSeconds` and `outPointSeconds` however it's awkward.  

In the `useEffect()` when `outPointSeconds <= inPointSeconds` I invoke `setOutPointSeconds(null)` to revert to a state where only `inPointSeconds` is set.  

But this immediately resets the `errorMessage` to `''` which in turn makes the error message disappear so that the user doesn't see it, and then the test ensuring the error message shows ends up failing because it disappears right away.

If we didn't invoke `setOutPointSeconds(null)` when `outPointSeconds <= inPointSeconds`, the invalid state would remain on the screen and the user would have to clear the range before starting again.  We agreed that when this error came up we wanted the `inPointSeconds` to remain set so the user could continue to complete the time range successfully, possibly by scrubbing to a correct position for the out-point.